### PR TITLE
feat(api): Add /auth/config endpoint

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Needed to determine initial login page state

### CODE REVIEW

File: src/sentry/api/endpoints/auth_config.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    def get(self, request, *args, **kwargs):
+        """
+        Get context required to show a login page. Registration is handled elsewhere.
+        """
+        if request.user.is_authenticated():
+            # if the user is a superuser, but not 'superuser authenticated' we
+            # allow them to re-authenticate to gain superuser status
+            if not request.user.is_superuser or is_active_superuser(request):
+                return self.respond_authenticated(request)
```
Comment (markstory) [inline src/sentry/api/endpoints/auth_config.py]: This is a code path I was not able to test during hackweek, might be worth a second look at it.

File: tests/sentry/api/endpoints/test_auth_config.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        assert response.status_code == 200
+        assert response.data['nextUri'] == '/auth/login/sentry/'
+
+    def test_superuser_is_not_redirected(self):
+        user = self.create_user('foo@example.com', is_superuser=True)
+        self.login_as(user)
+        response = self.client.get(self.path)
+
+        assert response.status_code == 200
+        assert 'nextUri' not in response.data
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_auth_config.py]: If the superuser isn't redirected will they simply stay on the login page?

Comment (EvanPurkhiser) [reply tests/sentry/api/endpoints/test_auth_config.py]: yup

File: src/sentry/api/endpoints/auth_config.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/api/endpoints/auth_config.py]: It's tested in the `test_superuser_is_not_redirected` test.

**Review by wedamija — APPROVED** [review]
Looks good. Can you document the code that needs to be removed from `frontend/auth_login.py`? Just want to make sure we remember to clean up once we're finished transitioning.

**Comment by EvanPurkhiser** [comment]
Yeah, I'll add some documentation around that in another PR.

**Pull request merged**